### PR TITLE
sql: support multiple databases with EXPLAIN ANALYZE (DEBUG)

### LINKED ISSUE #108265: stmtbundle: support multiple databases
Qualify the table name in the stats file with the database name, create each table in the schema.sql file in the correct database.

The need observed [here](https://github.com/cockroachlabs/support/issues/2478#issuecomment-1656508923).

Jira issue: CRDB-30398

### PR DESCRIPTION
**Pull request by rytaft**
Prior to this commit, statement bundles collected for cross-database queries could not be easily used with `cockroach debug statement-bundle recreate`. Due to not explicitly creating the relevant databases and the lack of fully-qualified names in the bundle, engineers would need to manually add these features to debug cross-database queries. This commit addresses these issues.

Fixes #108265

Release note: None

### CODE REVIEW

**Review by yuzefovich — APPROVED** [review]
Nice! I only have a few nits. Also there is one test that needs an adjustment, but otherwise :lgtm:

I think that it's worth backporting this change, so I added all the corresponding labels.

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115830#-:-Nl5Y0YeB-9_7RLoqLTZ:b-s3t8u9)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rytaft)

___
*[`pkg/sql/explain_bundle.go` line 541 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115830#-Nl5Wc9h-hjePSC3wgKv:-Nl5Wc9h-hjePSC3wgKw:blxndqd) ([raw file](https://github.com/cockroachdb/cockroach/blob/24abf15d8fe3f379b709982c1d645bcb0fe5d1be/pkg/sql/explain_bundle.go#L541)):*
> ```Go
> 		blankLine()
> 		if err := c.PrintCreateSequence(&buf, &sequences[i]); err != nil {
> 			b.printError(fmt.Sprintf("-- error getting schema for sequence %s: %v", sequences[i].String(), err), &buf)
> ```

nit: maybe also use `FQString` in error messages? Looks like there are 4 places like that.
___
*[`pkg/sql/explain_bundle_test.go` line 411 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115830#-Nl5XkNI982k4RVEh9J5:-Nl5XkNI982k4RVEh9J6:b-qtw12u) ([raw file](https://github.com/cockroachdb/cockroach/blob/24abf15d8fe3f379b709982c1d645bcb0fe5d1be/pkg/sql/explain_bundle_test.go#L411)):*
> ```Go
> // bundle contains the expected files. The expected files are passed as an
> // arbitrary number of strings; each string contains one or more filenames
> // separated by a space.
> ```

nit: might be worth adding a comment here that when `tableName` is non-empty, the function asserts that the substring equal to `tableName` argument is present in `schema.sql`. It doesn't have to be fully qualified name, but that is encouraged.

**Review by rytaft — COMMENTED** [review]
TFTR! I'll wait to bors until #115817 merges since I'll probably need to rebase.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115830#-:-Nl5mYMA5x9FC0BHW3Xw:b-mo8jw7)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @yuzefovich)

___
*[`pkg/sql/explain_bundle.go` line 541 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115830#-Nl5Wc9h-hjePSC3wgKv:-Nl5jgH50m8EAL7QGoFN:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/24abf15d8fe3f379b709982c1d645bcb0fe5d1be/pkg/sql/explain_bundle.go#L541)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: maybe also use `FQString` in error messages? Looks like there are 4 places like that.
</blockquote></details>

Done.
___
*[`pkg/sql/explain_bundle_test.go` line 411 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115830#-Nl5XkNI982k4RVEh9J5:-Nl5ks3_7EqJFdio1oS-:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/24abf15d8fe3f379b709982c1d645bcb0fe5d1be/pkg/sql/explain_bundle_test.go#L411)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: might be worth adding a comment here that when `tableName` is non-empty, the function asserts that the substring equal to `tableName` argument is present in `schema.sql`. It doesn't have to be fully qualified name, but that is encouraged.
</blockquote></details>

Done.

**Review by rytaft — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115830#-:-Nl_6dJm0vUI9xHjL3Ll:b-ydi9ay)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @yuzefovich)

**Pull request merged**
